### DISCUSSION TITLE
AppCast Generator: <sparkle:releaseNotesLink> issue?

### DISCUSSION THREAD

**Discussion by nicolatorpei** [NetSparkleUpdater/NetSparkle#426] [Q&A]
Hi, 
I have an issue with the appcast generator related to the `<sparkle:releaseNotesLink>`.  Could be related my local data file names or a small bug in the AppCastMaker.cs. This is my scenario:

- platform: Windows
- appcast parameters: `-a www -e exe -b www/builds -n "App Name" -u http://baseURL/appname/builds/ -p www/changelogs -l http://baseURL/appname/changelogs/ --overwrite-old-items --human-readable`
- local folders:
rootFolder/
├─ www/
│  ├─ builds/
│  │  ├─ build_1.0.exe
│  │  ├─ build_1.1.exe
│  ├─ changelogs/
│  │  ├─ 1.0.md
│  │  ├─ 1.1.md
│  ├─ appcast.xml

Running the generator I am not able to generate the appcast.xml filled with the tag `<sparkle:releaseNotesLink>` and the desired value of (for v. 1.0) `http://baseURL/appname/changelogs/1.0.md`.

So I debugged the AppCastMaker.cs (both last develop and last avalonia-preview branches commit) and I noticed that at row 204:

```
 // changelog stuff
var changelogFileName = productVersion + ".md";
var changelogPath = useChangelogs ? Path.Combine(_opts.ChangeLogPath, changelogFileName) : "";
var hasChangelogForFile = useChangelogs && File.Exists(changelogPath);
```

The `productVersion` string value has a lot of trailing spaces so the value of `File.Exists(changelogPath)` is always `false`.

Adding a `Trim()` to the `productName` string results in the expected behaviour.

```
 // changelog stuff - fixed
var changelogFileName = productVersion.Trim() + ".md";
var changelogPath = useChangelogs ? Path.Combine(_opts.ChangeLogPath, changelogFileName) : "";
var hasChangelogForFile = useChangelogs && File.Exists(changelogPath);
```

And I can have the appcast.xml with the `<sparkle:releaseNotesLink>` correctly filled for each build.

**Comment by Deadpikle** [comment]
Hi @nicolatorpei,

Thanks for the report and for doing your research ahead of time!

I tried to replicate your issue (https://github.com/NetSparkleUpdater/NetSparkle/commit/b26092089045dc719f30a8aa7f3dbedcaf6355a2). I believe that the whitespace is coming about due to lack of using the `--file-extract-version` flag to read versions from file names. Instead, the version is coming from your assembly, which may be the thing introducing the extra whitespace.

In any case, because I believe the extra whitespace is coming from the assembly version, I pushed the `.Trim()` call to there: https://github.com/NetSparkleUpdater/NetSparkle/commit/19803aa35be9b0cce466c8cbd6e32aa790eb21e5. That should fix it for you, I think, as well as for other functions that may utilize the version number.

I have pushed out a release for the app cast generator, 2.2.1.0, for this issue. Let me know if it works. Thanks! :)
